src/app/Plugins/Clock/Components/clock/clock.component.ts

```diff
@@ -36,14 +43,98 @@ export class ClockComponent implements OnInit {
       .subscribe((data: Clock[]) => this.clockData = [...data]);
   }
 
+  getSpecificAlarm() {
+    this.clockService.getSpecificAlarm().subscribe((data: Clock) => this.oneClockData = data);
+  }
+
   /*
     Sending a alarm to the service that sets it.
   */
-  setAlarm(hour: number, minute: number) {
-    this.newClockData.hour = hour;
-    this.newClockData.minute = minute;
-    this.clockService.setNewAlarm(this.newClockData).subscribe();
-    this.getAlarms();
+  setAlarm(day: number, hour: number, minute: number) {
```

Comment:
because of javascripte not checking types, here you declare type `number`, but you call the function https://github.com/AmyAssist/Amy-Web/pull/67/files#diff-c20410aebc41c1e9fb202945d381b1d5R41 with strings and hour will then be a string. Nobody ever checks that the argument matches the type.